web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/SelectSubnetIds.tsx

```diff
@@ -0,0 +1,173 @@
+  });
+
+  const { attempt, run } = useAttempt('processing');
+
+  function handleSelectSubnet(
+    subnet: Subnet,
+    e: React.ChangeEvent<HTMLInputElement>
+  ) {
+    if (e.target.checked) {
+      return onSelectedSubnetIds(currentSelectedGroups => [
+        ...currentSelectedGroups,
+        subnet.id,
+      ]);
+    } else {
+      onSelectedSubnetIds(selectedSubnetIds.filter(id => id !== subnet.id));
+    }
+  }
+
+  async function fetchSubnets({ refresh = false } = {}) {
+    run(() =>
+      integrationService
+        .fetchAwsSubnets(dbMeta.awsIntegration.name, clusterId, {
+          vpcId: dbMeta.vpcId,
+          region: dbMeta.awsRegion,
+          nextToken: tableData.nextToken,
+        })
+        .then(({ subnets, nextToken }) => {
+          const combinedSubnets = [...tableData.items, ...subnets];
+          setTableData({
+            nextToken,
+            fetchStatus: nextToken ? '' : 'disabled',
+            items: refresh ? subnets : combinedSubnets,
+          });
+          if (refresh) {
+            // Reset so user doesn't unintentionally keep a subnet
+            // that no longer exists upon refresh.
+            onSelectedSubnetIds([]);
+          }
+        })
+        .catch((err: Error) => {
+          const errMsg = getErrMessage(err);
+          emitErrorEvent(`fetch subnets error: ${errMsg}`);
+          throw err;
+        })
+    );
+  }
+
+  useEffect(() => {
+    fetchSubnets();
+  }, []);
+
+  return (
+    <>
+      <Flex alignItems="center" gap={1} mb={2}>
+        <Subtitle3>Select Subnets</Subtitle3>
+        <ToolTipInfo>
+          <Text>
+            A subnet has an outbound internet route if it has a route to an
+            internet gateway or a NAT gateway in a public subnet.
+          </Text>
+        </ToolTipInfo>
+      </Flex>
+
+      <Text mb={2}>
+        Select subnets to assign to the Fargate service that will be running the
+        database access agent. All of the subnets you select must have an
+        outbound internet route and a local route to the database subnets.
+      </Text>
+      {attempt.status === 'failed' && (
+        <>
+          <Flex my={3}>
+            <Icons.Warning size="medium" ml={1} mr={2} color="error.main" />
+            <Text>{attempt.statusText}</Text>
+          </Flex>
+          <ButtonBlueText ml={1} onClick={fetchSubnets}>
+            Retry
+          </ButtonBlueText>
+        </>
+      )}
+      {attempt.status === 'processing' && (
+        <Flex width="904px" justifyContent="center" mt={3}>
+          <Indicator />
+        </Flex>
+      )}
+      {attempt.status === 'success' && (
+        <Box mt={3}>
+          <SubnetIdPicker
+            region={dbMeta.awsRegion}
+            subnets={tableData.items}
+            attempt={attempt}
+            fetchNextPage={fetchSubnets}
+            fetchStatus={tableData.fetchStatus}
+            onSelectSubnet={handleSelectSubnet}
+            selectedSubnets={selectedSubnetIds}
```

Comment:
It'd be nice to have some kind of feedback as to how many or which subnets are selected. I just typed in something in the search bar, selected the only subnet that was matched and then removed my search query. The selected subnet was gone, since it's on page 2, and I wasn't sure if it's still selected or not.

Reply:
how about like this:

<img width="1036" alt="image" src="https://github.com/user-attachments/assets/1e2d110b-8f02-4dca-90d4-b4a11d84ed49">

Reply:
It would definitely solve the problem that I had. idk about the exact copy though, I'd  probably choose some variation of `Selected ${n} ${pluralize(n, 'security group')}` or `${n} ${pluralize(n, 'security group')} selected`.